src/BugsnagUnity/Delivery.cs

```diff
@@ -69,11 +72,108 @@ public void Deliver(IPayload payload)
             }
-            catch { }
+        }
+
+        private byte[] TruncateMetadata(IPayload payload)
+        {
+            var @event = payload.GetSerialisablePayload()["event"] as PayloadDictionary;
+            var metadata = @event["metaData"] as PayloadDictionary;
+            foreach (var section in metadata)
+            {
+                if (section.Key != "app" && section.Key != "device")
+                {
+                    TruncateStringsInMetadataSection(section.Value as Dictionary<string,object>);
+                }
+            }
+
+            var breadcrumbs = @event["breadcrumbs"] as Dictionary<string, object>[];
+            foreach (var crumb in breadcrumbs)
+            {
+                var crumbMetadata = crumb["metaData"] as Dictionary<string, object>;
+                TruncateStringsInMetadataSection(crumbMetadata);
+            }
+
+            return SerializePayload(payload);
+        }
+
+        private void TruncateStringsInMetadataSection(Dictionary<string, object> section)
+        {
+            foreach (var key in section.Keys.ToList())
+            {
+                var valueType = section[key].GetType();
+                if (valueType == typeof(string))
+                {
+                    var originalValue = section[key] as string;
+                    section[key] = CheckStringForTruncation(originalValue);
+                }
+                else if (valueType == typeof(string[]))
+                {
+                    var stringArray = section[key] as string[];
+                    for (int i = 0; i < stringArray.Length; i++)
+                    {
+                        stringArray[i] = CheckStringForTruncation(stringArray[i]);
+                    }
+                }
+                else if (valueType == typeof(List<string>))
+                {
+                    var stringArray = section[key] as List<string>;
+                    for (int i = 0; i < stringArray.Count; i++)
+                    {
+                        stringArray[i] = CheckStringForTruncation(stringArray[i]);
+                    }
+                }
+                else if (valueType == typeof(Dictionary<string, object>))
+                {
+                    TruncateStringsInMetadataSection(section[key] as Dictionary<string, object>);
+                }
+                else if (valueType == typeof(JsonArray))
+                {
+                    var array = ((JsonArray)section[key]);
+                    for (int i = 0; i < array.Count;i ++)
+                    {
+                        if (array[i].GetType() == typeof(string))
+                        {
+                            array[i] = CheckStringForTruncation((string)array[i]);
+                        }
+                    }
+                }
+            }
+        }
+
+        private string CheckStringForTruncation(string originalValue)
```

Comment:
nit: A slightly confusing method name as it actually truncates the string (not just checks) - `TruncateStringIfNecessary`?

Reply:
Nice 👍